docs/source/core_specification/schema_types.rst

```diff
@@ -1,3 +1,282 @@
+
+    directive @fold on FIELD
+
+    scalar Date
+
+    scalar DateTime
+
+    scalar Decimal
+
+    type Animal implements Entity {
+        name: String
+        alias: [String]
+        birthday: Date
+        net_worth: Decimal
+        in_Animal_ParentOf: [Animal]
+        out_Animal_ParentOf: [Animal]
+        in_Entity_Related: [Entity]
+        out_Entity_Related: [Entity]
+        out_Animal_OfSpecies: [Species]
+        out_Animal_PlaysWith: [Toy]
+    }
+
+    type Food implements Entity {
+        name: String
+        alias: [String]
+        in_Entity_Related: [Entity]
+        out_Entity_Related: [Entity]
+        in_Species_Eats: [Species]
+    }
+
+    type Species implements Entity {
+        name: String
+        alias: [String]
+        description: String
+        in_Animal_OfSpecies: [Animal]
+        in_Entity_Related: [Entity]
+        out_Entity_Related: [Entity]
+        in_Species_Eats: [Species]
+        out_Species_Eats: [Union__Food__Species]
+    }
+
+    type Toy {
+        name: String
+        in_Animal_PlaysWith: [Animal]
+    }
+
+    interface Entity {
+        name: String
+        alias: [String]
+        in_Entity_Related: [Entity]
+        out_Entity_Related: [Entity]
+    }
+
+    union Union__Food__Species = Food | Species
+
+
+.. note::
+
+    A GraphQL schema can be serialized with the :code:`print_schema` function from the
+    :code:`graphql.utils.schema_printer` module.
+
+
+Objects and Fields
+--------------------------
+
+The core components of a GraphQL schema are GraphQL object types. They conceptually represent the
+concrete vertices in the underlying database. For relational databases, we think of the tables as
+the concrete vertices.
+
+Lets go over a toy example of a GraphQL object type:
+
+.. code::
+
+    type Toy {
+        name: String
+        in_Animal_PlaysWith: [Animal]
+    }
+
+Here are some of the details:
+
+- :code:`name` is a **property field** that represents concretes data.
+- :code:`in_Animal_PlaysWith` is a **vertex field** representing an outbound edge.
+- :code:`String` is a built-in GraphQL scalar type.
+- :code:`[Animal]` is a GraphQL list type that represents an array of :code:`Animal`
+  objects.
+
+
+Directives
+----------
+
+In this section we'll go over how query directives are defined. For information on the available
+query directives and their semantics see :doc:`query directives <query_directives>`.
+
+Let's look at the :code:`@output` directive:
+
+.. code::
+
+   directive @output(out_name: String!) on FIELD
+
+- :code:`@output` defines the directive name.
+- :code:`out_name: String!` is a GraphQL argument. The :code:`!` indicates that it must not be null.
+- :code:`on FIELD` defines where the directive can be located. According to the definition, this
+  directive can only be located next to fields. The compiler might have additional restrictions
+  for where a query can be located. See :doc:`query directives <query_directives>` for more info.
```

Comment:
The entire `Directives` section is just explaining GraphQL syntax, and not anything specific to the compiler. How about removing the whole thing and just linking to the "real" GraphQL materials on directives instead?

Reply:
I think this section fits more nicely than the "real" GraphQL directive section. 
https://graphql.org/learn/queries/#directives

Reply:
You are completely right that this is explaining the GraphQL syntax, but by explaining the GraphQL syntax here, we have the benefit of only explaining the syntax that is relevant to the compiler.

Reply:
The "real" section can throw the reader off since it mentions directives we do not even mention in the schema.